### PR TITLE
Equipped Mobs proof of concept

### PR DESCRIPTION
**Pull request by MuteTiefling**
Simple proof of concept here. This gives us a config to randomly equip certain mobs with a set of gear on spawn.

Right now it will pick from a weighted list of gear sets and randomly enchant that equipment with a configurable level range if desired. 

I thought it could be a fun way to allow certain armor sets to be accessible before they can otherwise be crafted. Giving the player a chance to gear up with a bit of risk. Kind of like baby apotheosis bosses.

Probably still needs some tweaks, for example, I'd like to avoid overwriting actual apotheosis bosses with this, so I'll need to add in a check for those. I'll have to investigate the best way to determine that when there's a 1.19 release for that, however.

### CODE REVIEW

File: kubejs/server_scripts/base/entity/spawn/equipment.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+EntityEvents.spawned((event) => {
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/entity/spawn/equipment.js:1]: Perhaps rename the folder `spawn` -> `spawned` to match the event key?

**Review by NielsPilgaard — APPROVED** [review]
Love it! :D

File: kubejs/server_scripts/base/entity/spawn/equipment.js (hunk 1 above)
Comment (MuteTiefling) [reply kubejs/server_scripts/base/entity/spawn/equipment.js:1]: Oh good thinking

**Comment by MuteTiefling** [comment]
Will continue tweaking this over time. Gonna go ahead and merge for now to simplify things.

File: kubejs/server_scripts/base/entity/spawned/equipment.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
+EntityEvents.spawned((event) => {
+    if (!event.entity.isLiving()) {
+        return;
+    }
+
+    let mob = event.entity.type.split(':');
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/entity/spawned/equipment.js:6]: Could you perhaps make variables of `mob[0]` and `mob[1]`? I don't actually know what they are from reading this :P

Comment (MuteTiefling) [reply kubejs/server_scripts/base/entity/spawned/equipment.js:6]: Sure thing.

**Pull request merged**